packages/core/src/__integrationtests__/AccountLinking.spec.ts

```diff
@@ -233,98 +307,140 @@ describe('When there is an on-chain DID', () => {
+          keypair.address,
+        ])
         expect(
-          await AccountLinks.queryConnectedAccountsForDid(
-            newDid.uri,
-            ss58Format
-          )
-        ).toStrictEqual([keypair.address])
+          (
+            await api.query.didLookup.connectedAccounts.hash(
+              newDidChain,
+              paymentAccountChain
+            )
+          ).isEmpty
+        ).toBe(true)
         expect(
-          await AccountLinks.queryIsConnected(
-            newDid.uri,
-            paymentAccount.address
-          )
+          (
+            await api.query.didLookup.connectedAccounts.hash(
+              newDidChain,
+              keypairChain
+            )
+          ).isEmpty
         ).toBe(false)
-        expect(
-          await AccountLinks.queryIsConnected(newDid.uri, keypair.address)
-        ).toBe(true)
       })
       it('should be possible for the DID to remove the link', async () => {
-        const removeLinkTx = await AccountLinks.getLinkRemovalByDidExtrinsic(
-          keypair.address
+        const removeLinkTx = await api.tx.didLookup.removeAccountAssociation(
+          keypairChain
         )
         const signedTx = await Did.authorizeExtrinsic(
           newDid,
           removeLinkTx,
           newDidKey.sign,
           paymentAccount.address
         )
-        const balanceBefore = await Balance.getBalances(paymentAccount.address)
+        const balanceBefore = (
+          await api.query.system.account(paymentAccount.address)
+        ).data
         await submitExtrinsic(signedTx, paymentAccount)
 
         // Check that the deposit has been returned to the sender's balance.
-        const balanceAfter = await Balance.getBalances(paymentAccount.address)
+        const balanceAfter = (
+          await api.query.system.account(paymentAccount.address)
+        ).data
         expect(
           balanceBefore.reserved.sub(balanceAfter.reserved).toString()
         ).toStrictEqual(linkDeposit.toString())
         // Check that the link has been removed completely
         expect(
-          await AccountLinks.queryConnectedDidForAccount(paymentAccount.address)
-        ).toBeNull()
+          (await api.query.didLookup.connectedDids(paymentAccountChain)).isNone
+        ).toBe(true)
         expect(
-          await AccountLinks.queryConnectedDidForAccount(keypair.address)
-        ).toBeNull()
+          (await api.query.didLookup.connectedDids(keypairChain)).isNone
+        ).toBe(true)
         expect(
-          await AccountLinks.queryConnectedAccountsForDid(newDid.uri)
+          await api.query.didLookup.connectedAccounts.keys(newDidChain)
         ).toStrictEqual([])
         expect(
-          await AccountLinks.queryIsConnected(did.uri, paymentAccount.address)
-        ).toBe(false)
+          (
+            await api.query.didLookup.connectedAccounts.hash(
+              didChain,
+              paymentAccountChain
+            )
+          ).isEmpty
```

Comment:
this is another case that reads better if we use

```suggestion
            await api.query.didLookup.connectedAccounts(
              didChain,
              paymentAccountChain
            )
          ).isNone
```
Given that we do not know whether the `.hash` variant is indeed less costly somehow, i'd say we just go for the one that reads better

Reply:
Addressed in https://github.com/KILTprotocol/sdk-js/pull/634